src/main/java/org/elasticsearch/indices/cluster/IndicesClusterStateService.java

```diff
@@ -445,6 +469,7 @@ private boolean processMapping(String index, MapperService mapperService, String
             }
         } catch (Throwable e) {
             logger.warn("[{}] failed to add mapping [{}], source [{}]", e, index, mappingType, mappingSource);
```

Comment:
I think we get double logging now, no? maybe we should change this method to debug / trace (where we have the full mappingSource) and raise an exception which indicates which index and type, wrapping the underlying issue. That exception can be logged once in the catch clause of the calling method?